core/storage/trie/trie_storage_backend.hpp

```diff
@@ -15,12 +15,21 @@
 namespace kagome::storage::trie {
 
   /**
-   * Adapter for key-value storages that allows to hide keyspace separation
+   * Adapter for the trie node storage that allows to hide keyspace separation
    * along with root hash storing logic from the trie db component
    */
-  class TrieStorageBackend : public BufferStorage {
+  class TrieNodeStorageBackend : public BufferStorage {
    public:
-    ~TrieStorageBackend() override = default;
+    ~TrieNodeStorageBackend() override = default;
+  };
+
+  /**
+   * Adapter for the trie value storage that allows to hide keyspace separation
+   * along with root hash storing logic from the trie db component
+   */
+  class TrieValueStorageBackend : public BufferStorage {
+   public:
+    ~TrieValueStorageBackend() override = default;
```

Comment:
Remove `TrieNodeStorageBackend` and `TrieValueStorageBackend`.

Use `BufferStorage` directly (or new simple interface `TrieDb { get(Hash): Opt<Cow>; put(Hash, Bytes); rm(Hash) }`).

If you want strong type for injection, then don't inherit interface (because you can't use parent interface implementation `struct IDb{}; struct Db:IDb{}; struct IDb2:IDb{}; IDb2 idb2 = new Db{};`) but wrapper struct (`struct TrieDbNode { IDb _; }; struct TrieDbValue { IDb _; };` or `struct TrieDb { IDb node, value; }`)

Reply:
Why?